### DISCUSSION TITLE
If I were to make a library with `flutter_rust_bridge` to put on pub.dev, does the person downloading also need to set up `flutter_rust_bridge`?

### DISCUSSION THREAD

**Discussion by satvikpendem** [fzyzcjy/flutter_rust_bridge#1057] [Q&A] [answered]
I'm curious because I want to create an SDK as a library that uses `flutter_rust_bridge` but I'm not sure how feasible it is to ask the user to also follow all of the setup steps, especially if they are doing it on their production app and normally just expect to download libraries from pub.dev and use them in pure Dart. In other words, I'm using `flutter_rust_bridge` just as an implementation detail, the user wouldn't touch Rust code at all, they'd write in Dart which I'd then translate into a Rust call on the SDK side.

**Comment by fzyzcjy** [comment, marked as answer]
No! flutter_rust_bridge does nothing but generating a bunch of code that otherwise you as the programmer has to write by hand (e.g. bridge_generated.rs). Thus users do not need to do that.

If you want your users to download a precompiled .so file and use it, then they do not need to setup Rust toolchain. But if you want them to compile your Rust code from source, then they need to setup rust, which is indeed unrelated to flutter_rust_bridge, but yes that can be tedious.

This recently added doc my be related: https://cjycode.com/flutter_rust_bridge/library.html

Various packages using flutter_rust_bridge may also be helpful: e.g. https://github.com/Desdaemon/polars_dart. I remember there are a few more so maybe search a bit

**Reply by satvikpendem** [reply]
Awesome, let me take a look at those links and get back to you. We could also compile a separate Rust binary for each platform and distribute it, is that what you mean by .so?

**Reply by fzyzcjy** [reply]
I just mean binary (.so, .dylib, .a or whatever)
